artemis-cli/src/main/java/org/apache/activemq/artemis/cli/commands/util/SyncCalculation.java

```diff
@@ -54,92 +172,94 @@ public static long syncTest(File datafolder,
                                String fileName,
                                int maxAIO,
                                JournalType journalType) throws Exception {
-      SequentialFileFactory factory = newFactory(datafolder, fsync, journalType, blockSize * blocks, maxAIO);
-
+      final Supplier<? extends SyncIOCompletion> ioCallbackFactory;
```

Comment:
can this be just in the verbose section?

Reply:
essentially i would be expecting to see no diff in the code, except an if (verbose) statement with all the change only in that section.

Reply:
TBH I've provided a couple of fixes too  :P: like the one on the file size with `alignment != 1`.
I've tried to have a common logic while grouping the differences in order to have a simpler and maintainable code: probably that's the reason why you are seeing such differences on the source code.
I've already evaluated (ie JitWatch on the rescue) such abstraction won't affect the performance of the 2 cases, but if you think that is better to put everything under a big `if (verbose)` I can understand your reasons too: that's why having 6 eyes on it is more than wellcome IMO :)

Reply:
For me trying to understand the code just for a review and understand the differences/changes right now is a bit hard, as too much has moved, its not impossible just is making it tricky to be sure.

If you're not too much against it, it might be easier to do the "if", then it is simply cleaner / easier from a review POV to see what is changed / get comfortable that actually the old behaviour is as is.

Then anyhow as we get more confident it might the if statement always equates to true anyhow and the other side of the if block is removed entirely.

Reply:
> For me trying to understand the code just for a review and understand the differences/changes right now is a bit hard, as too much has moved, its not impossible just is making it tricky to be sure.

I think is normal: I've used the spitted view on github and it helps (vs the unified one), but probably it works just if you already know by heart the original code. Your review is a good way for me to have a more objective view of that so I will provide a cleaner separation of the 2 paths as a separate commit to be squashed if it will pass the reviews (by the end of the day; I've other stuff to do first :P).

Just for the sake of completeness, I will ask to @clebertsuconic too: he knows by heart the original code for sure! 
@clebertsuconic  wdyt about the last implementation? Have you found difficult to trace/understand how much has changed the not `--verbose` path vs the original?